presto-hive/src/test/java/io/prestosql/plugin/hive/parquet/ParquetDecimalScalingTest.java

```diff
@@ -0,0 +1,328 @@
+
+        JobConf jobConf = new JobConf();
+        jobConf.setEnum(COMPRESSION, UNCOMPRESSED);
+        jobConf.setBoolean(ENABLE_DICTIONARY, true);
+        jobConf.setEnum(WRITER_VERSION, PARQUET_2_0);
+
+        try {
+            FileSinkOperator.RecordWriter recordWriter = new TestMapredParquetOutputFormat(Optional.of(parquetSchema), true)
+                    .getHiveRecordWriter(
+                            jobConf,
+                            path,
+                            Text.class,
+                            false,
+                            tableProperties,
+                            () -> {});
+
+            Object row = inspector.create();
+            List<StructField> fields = ImmutableList.copyOf(inspector.getAllStructFieldRefs());
+
+            while (stream(iterators).allMatch(Iterator::hasNext)) {
+                for (int i = 0; i < fields.size(); i++) {
+                    Object value = iterators[i].next();
+                    inspector.setStructFieldData(row, fields.get(i), value);
+                }
+
+                ParquetHiveSerDe serde = new ParquetHiveSerDe();
+                serde.initialize(jobConf, tableProperties, null);
+                Writable record = serde.serialize(row, inspector);
+                recordWriter.write(record);
+            }
+
+            recordWriter.close(false);
+        }
+        catch (IOException | SerDeException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static void writeParquetDecimalsRecord(Path output, List<ParquetDecimalInsert> inserts)
+    {
+        List<String> fields = inserts.stream().map(ParquetDecimalInsert::schemaFieldDeclaration).collect(toImmutableList());
+        MessageType schema = parseMessageType(format("message record { %s; }", Joiner.on("; ").join(fields)));
+        List<ObjectInspector> inspectors = inserts.stream().map(ParquetDecimalInsert::getParquetObjectInspector).collect(toImmutableList());
+        List<String> columnNames = inserts.stream().map(ParquetDecimalInsert::getColumnName).collect(toImmutableList());
+        Iterator<?>[] values = inserts.stream().map(ParquetDecimalInsert::getValues).map(Iterable::iterator).toArray(Iterator[]::new);
+
+        createParquetFile(
+                output,
+                getStandardStructObjectInspector(columnNames, inspectors),
+                values,
+                schema,
+                Collections.singletonList("record"));
+    }
+
+    private static Properties createTableProperties(List<String> columnNames, List<ObjectInspector> objectInspectors)
+    {
+        Properties orderTableProperties = new Properties();
+        orderTableProperties.setProperty("columns", Joiner.on(',').join(columnNames));
+        orderTableProperties.setProperty("columns.types", Joiner.on(',').join(transform(objectInspectors, ObjectInspector::getTypeName)));
+        return orderTableProperties;
+    }
+
+    public static class ParquetDecimalInsert
```

Comment:
`private`

Reply:
importing static methods from it so protected